modules/calib3d/src/fisheye.cpp

```diff
@@ -475,17 +475,26 @@ void cv::fisheye::initUndistortRectifyMap( InputArray K, InputArray D, InputArra
 
         for( int j = 0; j < size.width; ++j)
         {
-            double x = _x/_w, y = _y/_w;
+            double u, v;
+            if( _w <= 0)
+            {
+                _x > 0 ? u = -std::numeric_limits<double>::infinity() : u = std::numeric_limits<double>::infinity();
+                _y > 0 ? v = -std::numeric_limits<double>::infinity() : v = std::numeric_limits<double>::infinity();
```

Comment:
Static analyzers doesn't like assignments in expressions. Could your use this form instead:
```
u = (_x > 0) ? -std::numeric_limits<double>::infinity() : std::numeric_limits<double>::infinity();
v = (_y > 0) ? ... : ...;
```

Reply:
Done